src/feature/RansacFeatureSetMatcher.cpp

```diff
@@ -147,14 +146,18 @@ double RansacFeatureSetMatcher::matchSets(const std::vector<InterestPoint *> &re
     }
     
     // Check if there are enough matches compared to the inlier probability 
-    if(double(possibleCorrespondences.size()) * m_inlierProbability < 2){  
-// 	std::cout << "Not enough possible correspondences for the inlier probability" << std::endl;
-	return 1e17;
+    if (double(possibleCorrespondences.size()) * m_inlierProbability < 2)
+    {
+        std::cout << "WARNING: not enough possible correspondences for the inlier probability" << std::endl;
     }
     
     boost::mt19937 rng;
     boost::uniform_smallint<int> generator(0, possibleCorrespondences.size() - 1);
     
+
+    double inlierProbability = m_adaptive ? 2.0 / possibleCorrespondences.size() : m_inlierProbability;
+    unsigned int iterations = ceil(log(1. - m_successProbability)/log(1. - inlierProbability * inlierProbability));
```

Comment:
If `possibleCorrespondences.size() == 2` and `m_adaptive == true`, you have to check that `log(1. - inlierProbability * inlierProbability)` doesn't do `log(0.0)`